pkg/chunked/compressor/compressor.go

```diff
@@ -383,7 +426,20 @@ func writeZstdChunkedStream(destFile io.Writer, outMetadata map[string]string, r
 	}
 	zstdWriter = nil
 
-	return internal.WriteZstdChunkedManifest(dest, outMetadata, uint64(dest.Count), metadata, level)
+	if err := tarSplitData.zstd.Flush(); err != nil {
```

Comment:
This branch doesn't close `tarSplitData.zstd`, and a number of branches above it don't close `zstdWriter`.

Reply:
there is a func for zstdWriter to clean it up if it is not closed (and thus set to nil afterwards):

```
	defer func() {
		if zstdWriter != nil {
			zstdWriter.Close()
		}
	}()
```

is that sufficient?

I've added the same pattern for tarSplitData